pkg/virt-v2v/conversion/conversion.go

```diff
@@ -0,0 +1,235 @@
+	"github.com/konveyor/forklift-controller/pkg/virt-v2v/utils"
+)
+
+type Conversion struct {
+	*config.AppConfig
+	// Disks to be converted
+	Disks []*Disk
+	// Used for injecting mock to the builder
+	CommandBuilder utils.CommandBuilder
+
+	fileSystem utils.FileSystem
+}
+
+func NewConversion(env *config.AppConfig) (*Conversion, error) {
+	conversion := Conversion{
+		AppConfig:      env,
+		CommandBuilder: &utils.CommandBuilderImpl{},
+		fileSystem:     &utils.FileSystemImpl{},
+	}
+
+	disks, err := conversion.getDisk()
+	if err != nil {
+		return nil, err
+	}
+	conversion.Disks = disks
+
+	return &conversion, nil
+}
+
+func (c *Conversion) getDisk() ([]*Disk, error) {
+	var disks []*Disk
+	diskPaths, err := filepath.Glob(config.FS)
+	if err != nil {
+		return nil, err
+	}
+	disksBlock, err := filepath.Glob(config.BLOCK)
+	if err != nil {
+		return nil, err
+	}
+	diskPaths = append(diskPaths, disksBlock...)
+	for _, path := range diskPaths {
+		disk, err := NewDisk(c.AppConfig, path)
+		if err != nil {
+			return nil, err
+		}
+		disks = append(disks, disk)
+	}
+	return disks, nil
+}
+
+// addCommonArgs adds a v2v arguments which is used for both virt-v2v and virt-v2v-in-place
+func (c *Conversion) addCommonArgs(cmd utils.CommandBuilder) error {
+	// Allow specifying which disk should be the bootable disk
+	if c.RootDisk != "" {
+		cmd.AddArg("--root", c.RootDisk)
+	} else {
+		cmd.AddArg("--root", "first")
+	}
+
+	// Add the mapping to the virt-v2v, used mainly in the windows when migrating VMs with static IP
+	if c.StaticIPs != "" {
+		for _, mac := range strings.Split(c.StaticIPs, "_") {
+			cmd.AddArg("--mac", mac)
+		}
+	}
+
+	// Adds LUKS keys, if they exist
+	if c.Luksdir != "" {
+		err := utils.AddLUKSKeys(c.fileSystem, cmd, c.Luksdir)
+		if err != nil {
+			return fmt.Errorf("error adding LUKS keys: %v", err)
+		}
+	}
+	if c.ExtraArgs != nil {
+		cmd.AddExtraArgs(c.ExtraArgs...)
+	}
+	return nil
+}
+
+func (c *Conversion) RunVirtV2VInspection() error {
+	v2vCmdBuilder := c.CommandBuilder.New("virt-v2v-inspector").
+		AddFlag("-v").
+		AddFlag("-x").
+		AddArg("-if", "raw").
+		AddArg("-i", "disk").
+		AddArg("-O", c.InspectionOutputFile)
+	err := c.addCommonArgs(v2vCmdBuilder)
+	if err != nil {
+		return err
+	}
+	for _, disk := range c.Disks {
+		v2vCmdBuilder.AddPositional(disk.Link)
+	}
+	v2vCmd := v2vCmdBuilder.Build()
+	v2vCmd.SetStdout(os.Stdout)
+	v2vCmd.SetStderr(os.Stderr)
+	return v2vCmd.Run()
+}
+
+func (c *Conversion) runVirtV2vInPlace() error {
+	v2vCmdBuilder := c.CommandBuilder.New("virt-v2v-in-place").
+		AddFlag("-v").
+		AddFlag("-x").
+		AddArg("-i", "libvirtxml")
+	err := c.addCommonArgs(v2vCmdBuilder)
+	if err != nil {
+		return err
+	}
+	v2vCmdBuilder.AddPositional(c.LibvirtDomainFile)
+
+	v2vCmd := v2vCmdBuilder.Build()
+	v2vCmd.SetStdout(os.Stdout)
+	v2vCmd.SetStderr(os.Stderr)
+	return v2vCmd.Run()
+}
+
+func (c *Conversion) addVirtV2vArgs(cmd utils.CommandBuilder) (err error) {
+	cmd.AddFlag("-v").
```

Comment:
Q: do we use "-on" always ? if we do, we can add it here and remove it elsewhere ?

Reply:
ah good point!